### ISSUE TITLE
Add an option for exercises (and modules?) which requires them to be completed before next exercises and/or materials can be opened

### ISSUE THREAD

**Issue by etanttila** [apluslms/a-plus#841]
Add an optional setting for exercises (and possibly also for modules) which requires them to be completed before next exercises and/or materials can be opened. This type of feature can be useful in several cases:
- Enrollment exercises (a questionnaire, an exercise or a "module 0" must be completed before the student can begin the course)
  - Would probably allow a larger variety of "enrollment exercises" than the current implementation
  - The current implementation of enrollment exercises may require revising anyway when the SISU integration takes care of enrollments
- Chapter or module feedback must be given before moving on to the next chapter or module
- Modules must be completed in order (or with enough points to pass before moving on to the next one)
- Multi-exercise projects must be completed in order (or with enough points)

Specific settings probably should be specified if this is developed, but it would probably be good to have settings for at least:
- are only exercises and submissions closed or also reading material
- which exercises or reading materials are prohibited before completion/passing this exercise (single ones or everything from one on)

**Comment by lainets** [comment]
Just to note that the systems (RevealRule) used for #830 should be usable in this case as well once it is done.

**Comment by Mankro** [comment]
A+ already has an old feature called threshold for this kind of purpose. The feature has never really been used and it can be only set up in the Django admin view.
https://github.com/apluslms/a-plus/blob/70513931a3f190552003143536d9238d1283bafb/threshold/models.py

**Comment by Mankro** [comment]
Very similar or a complete duplicate of https://github.com/apluslms/a-plus/issues/303.
